orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -218,7 +234,9 @@ def forward(self, *inputs, **kwargs):
             if self._device.type == 'cuda':
                 # Configure the InferenceSessions to use the specific GPU on which the model is placed.
                 providers = ["CUDAExecutionProvider", "CPUExecutionProvider"]
-                provider_options = [{"device_id": str(self._device.index)}]
+                provider_options = [{"device_id": str(self._device.index),
+                                     "cuda_external_alloc": str(self._torch_cuda_allocator.cuda_caching_allocator_raw_alloc_address()),
+                                     "cuda_external_free": str(self._torch_cuda_allocator.cuda_caching_allocator_raw_delete_address())}]
                 # Release CUDA cache used by PyTorch during exporter
```

Comment:
Should we add an internal `self._use_external_cuda_allocator` to enable this new behavior?

Reply:
and who will be setting that value? 

---
In reply to: [568132277](https://github.com/microsoft/onnxruntime/pull/6529#discussion_r568132277) [](ancestors = 568132277)

Reply:
we can keep it internally set with a default value of our choice and we manually change during execution. 
We do the same for `_save_onnx` already
This is intended for internal use and will not be present in the public api

When we have a proper option setting mechanism, we can refactor